modules/transactionpool/accept.go

```diff
@@ -10,273 +10,183 @@ import (
-	transactionPoolSize := len(encoding.Marshal(tp.transactionList))
-
-	if transactionPoolSize > TransactionPoolSizeLimit {
-		return ErrLargeTransactionPool
+	// Add the transaction set to the pool.
+	setID := TransactionSetID(crypto.HashObject(ts))
+	tp.transactionSets[setID] = ts
+	for _, diff := range cc.SiacoinOutputDiffs {
+		tp.knownObjects[ObjectID(diff.ID)] = struct{}{}
 	}
-	if transactionPoolSize > TransactionPoolSizeForFee {
-		var feeSum types.Currency
-		for _, fee := range t.MinerFees {
-			feeSum = feeSum.Add(fee)
-		}
-		if feeSum.Cmp(TransactionMinFee) < 0 {
-			return ErrLowMinerFees
-		}
+	for _, diff := range cc.FileContractDiffs {
+		tp.knownObjects[ObjectID(diff.ID)] = struct{}{}
 	}
-	return
-}
-
-// addTransactionToPool puts a transaction into the transaction pool, changing
-// the unconfirmed set and the transaction linked list to reflect the new
-// transaction.
-func (tp *TransactionPool) addTransactionToPool(t types.Transaction) {
-	// Apply each individual part of the transaction to the transaction pool.
-	tp.applySiacoinInputs(t)
-	tp.applySiacoinOutputs(t)
-	tp.applyFileContracts(t)
-	tp.applyFileContractRevisions(t)
-	tp.applyStorageProofs(t)
-	tp.applySiafundInputs(t)
-	tp.applySiafundOutputs(t)
-
-	// Add the transaction to the list of transactions.
-	tp.transactions[crypto.HashObject(t)] = struct{}{}
-	tp.transactionList = append(tp.transactionList, t)
+	for _, diff := range cc.SiafundOutputDiffs {
+		tp.knownObjects[ObjectID(diff.ID)] = struct{}{}
+	}
+	tp.transactionSetDiffs[setID] = cc
+	tp.databaseSize += len(encoding.Marshal(ts))
+	return nil
 }
 
 // AcceptTransaction adds a transaction to the unconfirmed set of
 // transactions. If the transaction is accepted, it will be relayed to
 // connected peers.
-func (tp *TransactionPool) AcceptTransaction(t types.Transaction) (err error) {
+func (tp *TransactionPool) AcceptTransactionSet(ts []types.Transaction) error {
 	id := tp.mu.Lock()
 	defer tp.mu.Unlock(id)
 
-	// Check that the transaction is not currently in the unconfirmed set.
-	txnHash := crypto.HashObject(t)
-	_, exists := tp.transactions[txnHash]
-	if exists {
-		return modules.ErrTransactionPoolDuplicate
-	}
-
-	// Check that the transaction against DoS prevention standards
-	err = tp.checkMinerFees(t)
+	err := tp.acceptTransactionSet(ts)
 	if err != nil {
-		return
-	}
-
-	// Check that the transaction is legal given the unconfirmed consensus set
-	// and the settings of the transaction pool.
-	err = tp.validUnconfirmedTransaction(t)
-	if err != nil {
-		return
+		return err
 	}
 
-	// Add the transaction to the pool, notify all subscribers, and broadcast
-	// the transaction.
-	tp.addTransactionToPool(t)
-	tp.updateSubscribers(modules.ConsensusChange{}, tp.transactionList, tp.unconfirmedSiacoinOutputDiffs())
-	go tp.gateway.Broadcast("RelayTransaction", t)
-	return
+	// Notify subscribers and broadcast the transaction set.
+	go tp.gateway.Broadcast("RelayTransactionSet", ts)
```

Comment:
this is a pretty breaking change. To smooth the transition, we will probably want to retain support for `"RelayTransaction"` for at least one version. Even then, new nodes won't be able to rely their transaction sets to old nodes.